pallets/ajuna-board/src/lib.rs

```diff
@@ -0,0 +1,202 @@
+			Turn = Self::PlayersTurn,
+			State = Self::GameState,
+		>;
+		/// Maximum number of players
+		#[pallet::constant]
+		type MaxNumberOfPlayers: Get<u32>;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub (super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub (super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Game has been created
+		GameCreated { board_id: T::BoardId, players: Vec<T::AccountId> },
+		/// Game has finished with the winner
+		GameFinished { winner: T::AccountId },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Not enough players for the game
+		NotEnoughPlayers,
+		/// Duplicate player found
+		DuplicatePlayer,
+		/// Too many players
+		TooManyPlayers,
+		/// Invalid state from game
+		InvalidStateFromGame,
+		/// Player not playing
+		NotPlaying,
+		/// Invalid turn played
+		InvalidTurn,
+		/// Invalid board
+		InvalidBoard,
+		/// Player already in game
+		PlayerAlreadyInGame,
+	}
+
+	type BoundedPlayersOf<T> =
+		BoundedVec<<T as frame_system::Config>::AccountId, <T as Config>::MaxNumberOfPlayers>;
+
+	type BoardGameOf<T> = BoardGame<<T as Config>::GameState, BoundedPlayersOf<T>>;
+
+	type PlayersOf<T> = BTreeSet<<T as frame_system::Config>::AccountId>;
+
+	#[pallet::storage]
+	pub type BoardStates<T: Config> = StorageMap<_, Identity, T::BoardId, BoardGameOf<T>>;
+
+	#[pallet::storage]
+	pub type PlayerBoards<T: Config> = StorageMap<_, Identity, T::AccountId, T::BoardId>;
+
+	#[pallet::storage]
+	pub type BoardIdCounter<T: Config> = StorageValue<_, T::BoardId, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create a new game with a set of players.
+		/// Players are unique and would not yet be in an existing game
+		#[pallet::weight(10_000)]
+		pub fn new_game(origin: OriginFor<T>, players: PlayersOf<T>) -> DispatchResult {
+			// TODO - This could be a whitelist based on a custom origin
+			// There is potentially more than one attack vector here as anyone could assign any
+			// account to a board and hence block them from playing in a legitimate game
+			// As this would be ran in L2 we may want to check that we are in L2??
+			let _ = ensure_signed(origin)?;
+			// Ensure we have players
+			ensure!(!players.is_empty(), Error::<T>::NotEnoughPlayers);
+
+			let player_len = players.len();
+			let players = BoundedPlayersOf::<T>::try_from(
+				players
+					.iter()
+					// Ensure that this player isn't already in a game
+					.filter(|player| !PlayerBoards::<T>::contains_key(player))
+					.cloned()
+					.collect::<Vec<T::AccountId>>(),
+			)
+			.map_err(|_| Error::<T>::TooManyPlayers)?;
+
+			// If we have new players this will be the same based on the filter
+			ensure!(player_len == players.len(), Error::<T>::PlayerAlreadyInGame);
+
+			let state = T::Game::init(&players).ok_or(Error::<T>::InvalidStateFromGame)?;
+			let next_board_id = BoardIdCounter::<T>::mutate(|counter| {
+				*counter += One::one();
```

Comment:
Is it safer to use `saturating_add` instead of `+`?

Reply:
It would be safer sure, not sure how long it would take to get there though